tools/InstallerSpec/InstallerSpec/Program.cs

```diff
@@ -0,0 +1,141 @@
+                {
+                    return serializer.Deserialize(fs) as DynamoInstallSpec;
+                }
+            }
+            catch (Exception ex) 
+            {
+                Console.WriteLine(string.Format("Failed to Create DynamoInstallSpec from file {0}", filePath));
+                Console.Write(ex.Message);
+            }
+
+            return null;
+        }
+
+        public void Save(string filePath)
+        {
+            var serializer = new XmlSerializer(typeof(DynamoInstallSpec), new[] { typeof(ModuleSpec) });
+            using (var fs = new FileStream(filePath, FileMode.Create, FileAccess.Write))
+            {
+                serializer.Serialize(fs, this);
+            }
+        }
+
+        public static IEnumerable<ModuleSpec> GetModules(DirectoryInfo dir)
+        {
+            foreach (var d in dir.EnumerateDirectories())
+            {
+                var modules = GetModules(d);
+                foreach (var m in modules)
+                {
+                    yield return m;
+                }
+            }
+
+            foreach (FileInfo item in dir.EnumerateFiles())
+            {
+                var info = FileVersionInfo.GetVersionInfo(item.FullName);
+                yield return new ModuleSpec() { Name = item.Name, FilePath = item.FullName.Replace(DynamoCoreVersion.BaseDirectory, @".\"), CopyForInstaller = true, DigitalSignature = Sign(info), Author = info != null ? info.CompanyName : "" };
+            }
+        }
+
+        private static bool Sign(FileVersionInfo info)
+        {
+            if(info != null && info.FileVersion != null)
+            {
+                return info.FileVersion.StartsWith(DynamoCoreVersion.BaseVersion);
+            }
+
+            return false;
+        }
+    }
+
+    class DynamoCoreVersion
+    {
+        public static string BaseVersion { get; private set; }
+        public static string BaseDirectory { get; private set; }
+        
+        public static void InitVersion(string binpath)
+        {
+            BaseDirectory = binpath;
+            if (!BaseDirectory.EndsWith(@"\"))
+                BaseDirectory += @"\";
+            string path = Path.Combine(binpath, "DynamoCore.dll");
+            if (!File.Exists(path))
+                return; //File is not available
+
+            var info = FileVersionInfo.GetVersionInfo(path);
+            BaseVersion = info.FileVersion.Substring(0, info.FileVersion.Length - 2); //drop last two characters
+        }
+    }
+
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            int nArgs = args.Length;
+            if(nArgs == 0)
+            {
+                Console.WriteLine("============== How to use this tool? ===========");
+                Console.WriteLine("Generates installer spec xml file to be used to create installer and digital signature based on a bin folder.");
+                Console.WriteLine("InstallerSpec.exe [bin folder] [xml file]");
```

Comment:
The default width for console output is 80 characters, perhaps the second line can be wrapped. Also, the help document should say this (because :

```
InstallerSpec.exe binfolder xmlfilepath

  binfolder     The folder in which all binaries reside. The search will 
                enumerate this folder as well as all sub-folders.

  xmlfilepath   The path to the output XML file path
```